chart/cas-cif/templates/jobs/terraform-apply.yaml

```
@@ -0,0 +1,73 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: terraform-apply
+  labels:
+    component: backend
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-install, pre-upgrade
+spec:
+  backoffLimit: 0
+  activeDeadlineSeconds: 900
+  template:
+    spec:
+      serviceAccountName: "terraform-kubernetes-service-account"
+      containers:
+        - name: terraform-apply
+          resources: {{ toYaml .Values.devops.resources | nindent 12 }}
+          image: "{{ .Values.devops.image.repository }}:{{ .Values.devops.sourceRepoImageTag | default .Values.devops.image.tag }}"
+          imagePullPolicy: {{ .Values.devops.image.pullPolicy }}
+          volumeMounts:
+            - mountPath: /etc/gcp
+              name: service-account-credentials-volume
+              readOnly: True
+            - mountPath: /etc/tf
+              name: terraform-backend-config-volume
+              readOnly: True
+            - name: tf-cache
+              mountPath: /working
+              readOnly: False
+            - name: terraform-modules
+              mountPath: /terraform
+              readOnly: False
+          env:
+            - name: TF_VAR_project_id
+              valueFrom:
+                secretKeyRef:
+                  name: gcp-credentials-secret
+                  key: gcp_project_id
+            - name: TF_VAR_openshift_namespace
+              value: {{ .Release.Namespace | quote }}
+            - name: TF_VAR_apps
+              value: '["cif-backups", "cif-documents"]'
+            - name: kubernetes_host
+              value: "https://api.silver.devops.gov.bc.ca:6443"
+            - name: GOOGLE_APPLICATION_CREDENTIALS
+              value: "/etc/gcp/credentials.json"
+          # Terraform was having an issue pulling kubernetes_host in as a TF_VAR, so we add it as a attribute to the command
+          command:
+            [
+              "/bin/sh",
+              "-c",
+              "cp -r /terraform/. /working && cd working && export TF_VAR_kubernetes_token=$( cat /var/run/secrets/kubernetes.io/serviceaccount/token ) && terraform init -backend-config=/etc/tf/gcs.tfbackend && terraform apply -var=\"kubernetes_host=$kubernetes_host\" -auto-approve",
+            ]
```

We might want to add a `set -euxo pipefail` to help debugging if the command fails
Helm sucks, and there's a slightly easier syntax to pass in bash scripts:

```suggestion
          command:
              - /bin/sh
              - -c
              - |
                  set -euo pipefail;
                  cp -r /terraform/. /working;
                  cd working;
                  export TF_VAR_kubernetes_token=$( cat /var/run/secrets/kubernetes.io/serviceaccount/token );
                  terraform init -backend-config=/etc/tf/gcs.tfbackend;
                  terraform apply -var=\"kubernetes_host=$kubernetes_host\" -auto-approve";
```

---

Thanks for the suggestion. I knew the long syntax wasn't the easiest to read, this makes it _much_ easier to understand! Changed with d068385a.